analytics/transform_test.go

```diff
@@ -315,6 +305,24 @@ func TestGetValue(t *testing.T) {
 	assert.NotNil(err)
 }
 
+func TestGetUnstructEventValue(t *testing.T) {
+	assert := assert.New(t)
+
+	// correct value unstruct field
+	unstructValue, err := fullEvent.GetUnstructEventValue("unstruct_event")
+	assert.Nil(err)
+	assert.Equal(unstructMap, unstructValue)
+}
+
+func TestGetContextValue(t *testing.T) {
+	assert := assert.New(t)
+
+	// correct value contexts
+	contextsValue, err := fullEvent.GetContextValue("contexts")
+	assert.Nil(err)
+	assert.Equal(contextsMap, contextsValue)
+}
+
```

Comment:
It'd be nice to see some additional tests to cover the more complete usage of these functions, such as the ones you describe in the PR:

```
GetContextValue(`derived_contexts`, `contexts_nl_basjes_yauaa_context_1`, `agentClass`)

GetUnstructEventValue(`unstruct_event`, `unstruct_event_com_snowplowanalytics_snowplow_add_to_cart_1`, `currency`, 0)
```

Reply:
Made tests more explicit

Reply:
Shouldn't there be tests for looking up more deeply nested data? I'd expect there to be a few more additional tests here for the more complex function calls that the new functions you've added are capable of doing. Please correct me if I'm missing something and that these tests do give more functionality coverage (rather than line coverage) than I can see.

Reply:
Nope that's not the case, contexts can have any data structure that UnstructEvent can have - it's not very common to have a lot of nesting but it happens

Reply:
@paulboocock @colmsnowplow I have added deep nested lookup in GetContextValue() using a ...path input parameter like in GetUnstructEventValue(). For the output maps, I have created a key using the path parameters converted to a string like this:

`[map[test.test2:map[test3:[works works2]]] map[test.test2:map[test3:works]]]`

Reply:
Fancy!